### PR TITLE
Ensure nested field support

### PR DESCRIPTION
**Pull request by bidoubiwa**
Add tests on nested fields to ensure it is still maintained in meilisearch

### CODE REVIEW

File: Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+public struct NestedBook: Codable, Equatable {
+  let id: Int
+  let title: String
+  let info: InfoNested
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift:4]: There is already a Info in Book.swift

[the remark this answers is not in the document]
Comment (brunoocasali) [reply Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift:4]: I mean something like this:
```suggestion
  let info: Info
```

Use this name  :)

**Review by brunoocasali — APPROVED** [review]
💯 I left a comment about the `InfoNested` to `Info` naming, but is up to you!

File: Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift:4]: Yes but there would be a conflict with the `public struct Info` declared in Books.swift

```
meilisearch-swift/Tests/MeiliSearchIntegrationTests/Support/Book.swift:45:15: error: invalid redeclaration of 'Info'
```

[the remark this answers is not in the document]
Comment (brunoocasali) [reply Tests/MeiliSearchIntegrationTests/Support/NestedBook.swift:4]: Oh I see that now, let's keep this way then 😅

**Pull request merged**